doc/openocd.texi

```diff
@@ -10812,6 +10812,14 @@ When utilizing version 0.11 of the RISC-V Debug Specification,
 and DBUS registers, respectively.
 @end deffn
 
+@deffn {Command} {riscv smp} [on|off]
+Display/set the current SMP mode
+@end deffn
+
+@deffn {Command} {riscv smp_gdb} [core_id]
+Display/set the current core displayed in GDB
+@end deffn
```

Comment:
Given the discussion you had with Jan, can you mention that it's generally preferable to use `-rtos hwthread` instead of using these commands to set SMP mode or current core? It makes sense to use the existing commands for detection, but I imagine this opens up more corner cases that I don't really want to test, because using this SMP mechanism (specifically for RISC-V) should be considered deprecated.

Reply:
Thank you for the summary above.

My understanding was that it is unusual to use SMP groups (defined by `target smp ...`) without `-rtos hwthread`. 

I even thought it was deprecated, but I was probably mistaken here, as I can't find a mention of this deprecation the doc or mailing list. Sorry for this misleading comment.

@aap-sc My questions would be: 

- Do you actively use the SMP groups (defined by `target smp ...`) without `-rtos hwthread`?
- If yes, are you using it also with GDB this way? Or only without GDB, that is in scripts or in the telnet console?

Thank you.

Reply:
> My understanding was that it is unusual to use SMP groups (defined by target smp ...) without -rtos hwthread.

This my understanding too. 

> Do you actively use the SMP groups (defined by target smp ...) without -rtos hwthread?

No. Though, I see such configurations from time to time, but most likely this is a consequence of copy-pasting erroneous configuration - not the conscious choice of the user.

Reply:
@aap-sc That's an excellent description. I'm concerned that people will see `riscv smp` and confuse it with `target smp`. Most users will never want to use `riscv smp`. So can you add some kind of explanation in the documentation that makes it clear that this is a distinct command from `target smp`, and that most users will want the latter and not the former.

Something like:
> `riscv smp` allows fine control over which targets are in the SMP group exposed to gdb. For most use cases, `target smp` is recommended.

Reply:
Tried to address... @timsifive , @JanMatCodasip would you kindly take a look once again?